packages/common/src/lib/tool/shared/toolbox.ts

```diff
@@ -118,6 +119,13 @@ export class Toolbox {
     this.activateTool(previous);
   }
 
+  getCurrentPreviousToolName(): [string, string] {
+    const [previous, current] = this.toolHistory.slice(
+      Math.max(this.toolHistory.length - 2, 0)
+    );
```

Comment:
Qu'arrive t'il si on a un seul élément dans le tableau? Le current est attribué au previous, est-ce quelque chose de possible?

Shortcut:
```suggestion
    const [previous, current] = this.toolHistory.slice(-2);
```

Reply:
oui  le current est attribué au previous si il y a un seul élément dans le tableau.

Reply:
Dans ce cas ce n'est pas le previous mais le current! On devrait géré ce cas de figure.

Reply:
ok